### PR TITLE
Add Python testing infrastructure

### PR DESCRIPTION
**Pull request by jkfran**
## Done

- Have a `tests` folder
- Have at least one test in the `tests` folder, based on `unittest`, to test that the `/` homepage returns a `200` response
- `./run test` run `lint-python` and `test-python`
- `./run lint-python` run `flake8` and `black --check --line-length 79` against all Python code (including tests)
- `./run test-python` run the `tests`
- Set up CircleCI with a job called `test-python`, which run the tests
- Added codecov integration on CircleCI

## QA

- Check out this feature branch
- Run the command `./run test`, it should end correctly.
- Run the command `./run lint-python`, it should end correctly.
- Run the command `./run test-python`, it should end correctly.

### CODE REVIEW

File: tests/test_routes.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        app.testing = True
+        self.client = app.test_client()
+
+    def test_homepage_no_session(self):
+        """
+        When given the index URL without session,
+        we should return a 302 status code redirection
+        """
+
+        self.assertEqual(self.client.get("/").status_code, 302)
```
Comment (jkfran) [inline tests/test_routes.py]: @nottrobin looks like this project need SSO authentication. Do you think I should try to create an SSO session in the client and then test to get a 200 status code?

**Comment by nottrobin** [comment]
@jkfran please update this branch when you get a chance =)

**Comment by jkfran** [comment]
@nottrobin Done, what do you think about my previous questions about the SSO session?

File: tests/test_routes.py (hunk 1 above)
Comment (nottrobin) [reply tests/test_routes.py]: No I think that's overkill for now. I think testing for a 302 response from `/` is fine.

Though I don't think it's necessary right now, if you really want to expand on this with better SSO verification, that would actually be awesome. But let's do it in a follow-up PR so we can merge this one now.

**Pull request merged**